src/exporters/file/task.rs

```diff
@@ -0,0 +1,458 @@
+use crate::bounded_channel::BoundedReceiver;
+use crate::exporters::file::config::FileExporterConfig;
+use crate::exporters::file::json::JsonExporter;
+use crate::exporters::file::parquet::{LogRecordRow, MetricRow, ParquetExporter, SpanRow};
+use crate::exporters::file::{FileExporterError, Result};
+use crate::init::file_exporter::FileExporterFormat;
+use chrono::Utc;
+use opentelemetry_proto::tonic::logs::v1::ResourceLogs;
+use opentelemetry_proto::tonic::metrics::v1::ResourceMetrics;
+use opentelemetry_proto::tonic::trace::v1::ResourceSpans;
+use std::path::Path;
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, info};
+
+/// Generic trait for file exporters that can handle different data formats
+trait TypedFileExporter {
+    /// The type used to represent span data in this exporter
+    type SpanData: Clone;
+    /// The type used to represent metric data in this exporter
+    type MetricData: Clone;
+    /// The type used to represent log data in this exporter
+    type LogData: Clone;
+
+    /// Convert ResourceSpans to the exporter's span data format
+    fn convert_spans(&self, resource_spans: &ResourceSpans) -> Result<Vec<Self::SpanData>>;
+    /// Convert ResourceMetrics to the exporter's metric data format
+    fn convert_metrics(&self, resource_metrics: &ResourceMetrics) -> Result<Vec<Self::MetricData>>;
+    /// Convert ResourceLogs to the exporter's log data format
+    fn convert_logs(&self, resource_logs: &ResourceLogs) -> Result<Vec<Self::LogData>>;
+
+    /// Export span data to a file
+    fn export_spans(&self, data: &[Self::SpanData], path: &Path) -> Result<()>;
+    /// Export metric data to a file
+    fn export_metrics(&self, data: &[Self::MetricData], path: &Path) -> Result<()>;
+    /// Export log data to a file
+    fn export_logs(&self, data: &[Self::LogData], path: &Path) -> Result<()>;
+
+    /// Get the file extension for this exporter
+    fn file_extension(&self) -> &'static str;
+}
+
+/// Implementation of TypedFileExporter for ParquetExporter
+impl TypedFileExporter for ParquetExporter {
```

Comment:
Can you move these TypedFileExporter implementations into the files with the concrete types (ParquetExporter and JsonExporter)?

Reply:
Do you look at this?